packages/pages/src/scaffold/modules/generate.test.ts

```diff
@@ -0,0 +1,27 @@
+import { describe, it, expect } from "vitest";
+import { validateModuleNameFormat } from "./generate.js";
+
+describe("validateModuleNameFormat", () => {
+  const testCases = [
+    { moduleName: "foo123", expected: true },
+    { moduleName: "foo-bar123", expected: true },
+    { moduleName: "foo_bar", expected: true },
+    { moduleName: "foo_bar", expected: true },
+    { moduleName: "foo$bar", expected: true },
+    { moduleName: "foobar-_$", expected: true },
```

Comment:
What does this get turned into?

Reply:
foobar-_$ gets turned into Foobar_$ for the exported Module const.